Error building on MSVC

From Windows weekly verification ( https://github.com/apache/arrow-nanoarrow/actions/runs/5914290765/job/16039460039#step:11:314 )

```
   ClCompile:
    C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX64\x64\CL.exe /c /I"D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src" /I"D:\a\arrow-nanoarrow\arrow-nanoarrow\nanoarrow-verify-tmp\build\generated" /Zi /nologo /W3 /WX- /diagnostics:column /Od /Ob0 /D _MBCS /D WIN32 /D _WINDOWS /D NANOARROW_DEBUG /D "CMAKE_INTDIR=\"Debug\"" /Gm- /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++17 /Fo"buffer_test.dir\Debug\\" /Fd"buffer_test.dir\Debug\vc143.pdb" /external:W0 /Gd /TP /errorReport:queue  /external:I "D:/a/arrow-nanoarrow/arrow-nanoarrow/nanoarrow-verify-tmp/build/_deps/googletest-src/googletest/include" /external:I "D:/a/arrow-nanoarrow/arrow-nanoarrow/nanoarrow-verify-tmp/build/_deps/googletest-src/googletest" "D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src\nanoarrow\buffer_test.cc"
    buffer_test.cc
  D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src\nanoarrow\buffer_test.cc(297,17): error C2131: expression did not evaluate to a constant [D:\a\arrow-nanoarrow\arrow-nanoarrow\nanoarrow-verify-tmp\build\buffer_test.vcxproj]
  D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src\nanoarrow\buffer_test.cc(297,17): message : failure was caused by a read of a variable outside its lifetime [D:\a\arrow-nanoarrow\arrow-nanoarrow\nanoarrow-verify-tmp\build\buffer_test.vcxproj]
  D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src\nanoarrow\buffer_test.cc(297,17): message : see usage of 'n_values' [D:\a\arrow-nanoarrow\arrow-nanoarrow\nanoarrow-verify-tmp\build\buffer_test.vcxproj]
  D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src\nanoarrow\buffer_test.cc(298,20): error C2131: expression did not evaluate to a constant [D:\a\arrow-nanoarrow\arrow-nanoarrow\nanoarrow-verify-tmp\build\buffer_test.vcxproj]
  D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src\nanoarrow\buffer_test.cc(298,20): message : failure was caused by a read of a variable outside its lifetime [D:\a\arrow-nanoarrow\arrow-nanoarrow\nanoarrow-verify-tmp\build\buffer_test.vcxproj]
  D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src\nanoarrow\buffer_test.cc(298,20): message : see usage of 'n_values' [D:\a\arrow-nanoarrow\arrow-nanoarrow\nanoarrow-verify-tmp\build\buffer_test.vcxproj]
  D:\a\arrow-nanoarrow\arrow-nanoarrow\src\src\nanoarrow\buffer_test.cc(302,21): fatal  error C1903: unable to recover from previous error(s); stopping compilation [D:\a\arrow-nanoarrow\arrow-nanoarrow\nanoarrow-verify-tmp\build\buffer_test.vcxproj]
```

--- a/src/nanoarrow/buffer_test.cc
+++ b/src/nanoarrow/buffer_test.cc
@@ -293,8 +293,8 @@ void TestArrowBitmapUnpackUnsafe(const uint8_t* bitmap, std::vector<int8_t> expe
 
 TEST(BitmapTest, BitmapTestBitmapUnpack) {
   uint8_t bitmap[3];
-  int64_t n_values = sizeof(bitmap) * 8;
-  int8_t result[n_values];
+  int8_t result[sizeof(bitmap) * 8];
+  int64_t n_values = sizeof(result);
   int32_t result32[n_values];
 
   // Basic test of a validity buffer that is all true

